addons/hr_expense/models/product_product.py

```diff
@@ -0,0 +1,19 @@
+from odoo import api, fields, models, _
+
+
+class ProductProduct(models.Model):
+    _inherit = "product.product"
+
+    standard_price_update_warning = fields.Char(compute="_compute_standard_price_update_warning")
+
+    @api.depends('standard_price')
+    def _compute_standard_price_update_warning(self):
+        undone_expenses = self.env['hr.expense'].search([('state', 'in', ['draft', 'reported', 'approved']), ('product_id', 'in', self.ids)])
+        for product in self:
+            product.standard_price_update_warning = False
+            # The following list is composed of all the expenses' unit_amounts that should NOT trigger a warning.
+            # Those are the amounts of any undone expense using this product and 0.0 which is the default unit_amount.
+            unit_amounts_to_avoid = [expense.unit_amount if expense.product_id.id == product.id else False for expense in undone_expenses]+[0.0]
+            if undone_expenses and not (product.standard_price in unit_amounts_to_avoid):
+                product.standard_price_update_warning = _("""There are unposted expenses linked to this category.
```

Comment:
when a translatable string is modified or added in stable, the change must be reflected in the .pot file :point_up: